src/vision.cs

```diff
@@ -3872,4 +4223,265 @@ interface VNDetectDocumentSegmentationRequest {
+		NativeHandle Constructor ([NullAllowed] VNRequestCompletionHandler completionHandler);
+
+		[Export ("supportedJointNamesAndReturnError:")]
+		[return: NullAllowed]
+		[return: BindAs (typeof (VNHumanBodyPose3DObservationJointName []))]
+		NSString [] GetSupportedJointNames ([NullAllowed] out NSError error);
+
+		[Export ("supportedJointsGroupNamesAndReturnError:")]
+		[return: NullAllowed]
+		[return: BindAs (typeof (VNHumanBodyPose3DObservationJointsGroupName []))]
+		NSString [] GetSupportedJointsGroupNames ([NullAllowed] out NSError error);
+
+		[NullAllowed, Export ("results", ArgumentSemantic.Copy)]
+		VNHumanBodyPose3DObservation [] Results { get; }
+	}
+
+	[TV (17, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface VNPoint3D : NSCopying, NSSecureCoding {
+		[Export ("initWithPosition:")]
+		[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+		[DesignatedInitializer]
+		NativeHandle Constructor (Vector4 position);
+
+		[Export ("position")]
+		Vector4 Position {
+			[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+			get;
+		}
+	}
+
+	[TV (17, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (VNPoint3D))]
+	[DisableDefaultCtor]
+	interface VNRecognizedPoint3D {
+		[Export ("initWithPosition:")]
+		[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+		[DesignatedInitializer]
+		NativeHandle Constructor (Vector4 position);
+
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[Field ("VNRecognizedPoint3DGroupKeyAll")]
+		NSString GroupKeyAll { get; }
+	}
+
+	[TV (17, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (VNObservation))]
+	[DisableDefaultCtor]
+	interface VNRecognizedPoints3DObservation {
+		[Export ("availableKeys", ArgumentSemantic.Copy)]
+		NSString [] AvailableKeys { get; }
+
+		[Export ("availableGroupKeys", ArgumentSemantic.Copy)]
+		NSString [] AvailableGroupKeys { get; }
+
+		[Export ("recognizedPointForKey:error:")]
+		[return: NullAllowed]
+		VNRecognizedPoint3D GetRecognizedPoint (NSString pointKey, [NullAllowed] out NSError error);
+
+		[Export ("recognizedPointsForGroupKey:error:")]
+		[return: NullAllowed]
+		NSDictionary<NSString, VNRecognizedPoint3D> GetRecognizedPoints (NSString groupKey, [NullAllowed] out NSError error);
+	}
+
+	[TV (17, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (VNRecognizedPoint3D))]
+	[DisableDefaultCtor]
+	interface VNHumanBodyRecognizedPoint3D {
+		[Export ("initWithPosition:")]
+		[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+		[DesignatedInitializer]
+		NativeHandle Constructor (Vector4 position);
+
+		[Export ("localPosition")]
+		Vector4 LocalPosition {
+			[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+			get;
+		}
+
+		[Export ("parentJoint")]
+		string ParentJoint { get; }
```

Comment:
We should look at the docs, this sounds like one of the smart enums if not probably NSString is a better choice here.

```suggestion
		NSString ParentJoint { get; }
```

Reply:
@dalexsoto @mandel-macaque yep good catch, it does return the enum https://developer.apple.com/documentation/vision/vnhumanbodyrecognizedpoint3d/4173215-parentjoint?language=objc